Sources/HAP/Controllers/PairSetupController.swift

```diff
@@ -32,6 +56,34 @@ class PairSetupController {
             throw Error.invalidPairingMethod
         }
 
+        // If the accessory is already paired it must respond with
+        // Error_Unavailable
+        if device.isPaired {
+            return [
+                .state: Data(bytes: [PairSetupStep.startResponse.rawValue]),
+                .error: Data(bytes: [PairError.unavailable.rawValue])
+            ]
+        }
+
+        // If the accessory has received more than 100 unsuccessful
+        // authentication attempts it must respond with
+        // Error_MaxTries
+        // TODO
+
+        // If the accessory is currently performing a Pair Setup operation with
+        // a different controller it must respond with
+        // Error_Busy
+        if device.lastPairingEvent == .pairingStarted ||
+            device.lastPairingEvent == .pairingVerified {
```

Comment:
Having a state machine, we wouldn't have to have all such logic throughout various controllers. We would ask the state machine if we can change into "pairing started", which would fail if we're not in the correct state (see also state diagram).